load-tests/load_tests/runner.py

```diff
@@ -24,20 +24,30 @@ def __init__(
 
     def run(self, scenario: Scenario, client: Client, session_id: str) -> None:
         interactions: List[Interaction] = self._resolve_interactions(scenario)
-
         start_time = time.time()
+        accumulated_messages = []
 
         @client.event
         def bot_uttered(data):
             response_time = int((time.time() - start_time) * 1000)  # epochs to ms
-            request_success.fire(
-                request_type="socketio received",
-                name=EVENT_BOT_UTTERED,
-                response_time=response_time,
-                response_length=len(str(data)),
-            )
+            accumulated_messages.append(data)
+
             if len(interactions) > 0:
-                user_uttered(interactions.pop(0))
+                expected_bot = self._interaction_loader.render_bot_turn(
+                    interactions[0].bot
+                )
+                if len(accumulated_messages) == len(expected_bot):
+                    request_success.fire(
+                        request_type="socketio received",
+                        name=EVENT_BOT_UTTERED,
+                        response_time=response_time,
+                        response_length=len(str(data)),
+                    )
+                    accumulated_messages.clear()
+                    user_uttered(interactions.pop(0))
+
+            elif len(accumulated_messages) > 0:
```

Comment:
not sure I get the purpose of this condition. Why should `len(accumulated_messages) > 0` before we disconnect? Could there be other messages pending? Shouldn't the be included in the expected interactions?

Reply:
We are expecting more messages after the final user_input, I wait until the server responds to disconnect else the server would produce an error. Not the optimal solution but the other solution involves adapting the tests to socketio which will take a lot more work.